src/view/newtListMenu.cpp

```diff
@@ -6,3 +6,76 @@
 // This file is empty since the method is now a template
 
 #include <cloysterhpc/view/newt.h>
+#include <newt.h>
+#include <ranges>
+
+static std::vector<std::string> retrieveListSelectedItems(newtComponent list)
+{
+    int selectCount = 0;
+    char** selectItems
+        = reinterpret_cast<char**>(newtListboxGetSelection(list, &selectCount));
+    auto temp = std::vector(selectItems, selectItems + selectCount)
+        | std::views::transform([](char* c) { return std::string { c }; });
+    std::vector<std::string> ret(std::begin(temp), std::end(temp));
+    return ret;
+}
+
+std::pair<int, std::vector<std::string>> Newt::multipleSelectionMenu(
+    const char* title, const char* message, const char* help,
+    std::vector<std::tuple<std::string, std::string, bool>> items)
+{
+    auto* form = newtForm(nullptr, nullptr, 0);
+    newtFormSetWidth(form, m_suggestedWidth);
+
+    auto* label = newtLabel(1, 1, message);
+    auto stitle = std::string { title };
+
+    auto* list = newtListbox(1, 3, m_maxListHeight, NEWT_FLAG_MULTIPLE);
+
+    for (const auto& [key, item, enabled] : items) {
+        newtListboxAppendEntry(list, item.c_str(), (void*)key.c_str());
+
+        if (enabled)
+            newtListboxSelectItem(list, key.c_str(), NEWT_FLAGS_SET);
+    }
+
+    newtGrid grid = newtCreateGrid(1, 3);
+    newtComponent bOk, bCancel, bHelp;
+    newtGrid buttonGrid = newtButtonBar(const_cast<char*>(TUIText::Buttons::ok),
+        &bOk, const_cast<char*>(TUIText::Buttons::cancel), &bCancel,
+        const_cast<char*>(TUIText::Buttons::help), &bHelp, NULL);
+    newtGridSetField(grid, 0, 0, NEWT_GRID_COMPONENT, label, 1, 1, 0, 0, 0,
+        NEWT_GRID_FLAG_GROWX);
+    newtGridSetField(grid, 0, 1, NEWT_GRID_COMPONENT, list, 1, 1, 0, 0, 0,
+        NEWT_GRID_FLAG_GROWX | NEWT_GRID_FLAG_GROWY);
+    newtGridSetField(grid, 0, 2, NEWT_GRID_SUBGRID, buttonGrid, 0, 1, 0, 0, 0,
+        NEWT_GRID_FLAG_GROWX);
+    newtGridWrappedWindow(grid, const_cast<char*>(stitle.c_str()));
```

Comment:
Hmmm do we need creating a local copy?

Reply:
Of what? Of the string?

If you are talking about `stitle`, then _yes_. This function (`newtGridWrappedWindow`) is not const-correct, and I created a "sacrificial" copy to preserve the original string, in the rare case this function _does_ change the string

Reply:
This will violate constness of the original value

Reply:
Take a look on the comment on the `auto` implicit conversion.